Libraries/acdriver/Sources/CompileActionImageSet.cpp

```diff
@@ -29,183 +29,172 @@
+    if (png_struct_ptr == NULL) {
         result->normal(
             Result::Severity::Error,
             "png_create_read_struct returned error",
-            std::string(filename));
+            filename);
         return false;
     }
 
-    info_struct_ptr = png_create_info_struct(png_struct_ptr);
-    if (!info_struct_ptr) {
-        png_destroy_read_struct (&png_struct_ptr, NULL, NULL);
+    png_info *info_struct_ptr = png_create_info_struct(png_struct_ptr);
+    if (info_struct_ptr == NULL) {
+        png_destroy_read_struct(&png_struct_ptr, NULL, NULL);
         result->normal(
             Result::Severity::Error,
             "png_create_info_struct returned error",
-            std::string(filename));
+            filename);
         return false;
     }
 
     if (setjmp(png_jmpbuf(png_struct_ptr))) {
-        png_destroy_read_struct (&png_struct_ptr, &info_struct_ptr, NULL);
+        png_destroy_read_struct(&png_struct_ptr, &info_struct_ptr, NULL);
         result->normal(
             Result::Severity::Error,
             "setjmp/png_jmpbuf returned error",
-            std::string(filename));
+            filename);
         return false;
     }
 
-    unsigned char *contents_ptr = static_cast<unsigned char*>(contents.data());
+    unsigned char *contents_ptr = static_cast<unsigned char *>(contents.data());
     png_set_read_fn(png_struct_ptr, &contents_ptr, png_user_read_data);
 
     png_read_info(png_struct_ptr, info_struct_ptr);
 
     png_uint_32 width, height;
     int bit_depth, color_type, interlace_method, compression_method, filter_method, channels;
-    if (!png_get_IHDR (png_struct_ptr, info_struct_ptr,
-        &width, &height, &bit_depth, &color_type, &interlace_method, &compression_method, &filter_method)) {
-        png_destroy_read_struct (&png_struct_ptr, &info_struct_ptr, NULL);
+    if (!png_get_IHDR(png_struct_ptr, info_struct_ptr, &width, &height, &bit_depth, &color_type, &interlace_method, &compression_method, &filter_method)) {
+        png_destroy_read_struct(&png_struct_ptr, &info_struct_ptr, NULL);
         result->normal(
             Result::Severity::Error,
-            "setjmp/png_jmpbuf returned error",
-            std::string(filename));
+            "failed to read PNG header",
+            filename);
         return false;
     }
 
-    /* Convert pixels to RGBA or GA8 */
+    /* Convert pixels to RGBA or GA8. */
     if (color_type == PNG_COLOR_TYPE_PALETTE) {
         png_set_palette_to_rgb(png_struct_ptr);
         png_set_bgr(png_struct_ptr);
         png_set_filler(png_struct_ptr, 0xff, PNG_FILLER_AFTER);
-    }
-
-    if (color_type == PNG_COLOR_TYPE_GRAY &&
-        bit_depth < 8) {
+    } else if (color_type == PNG_COLOR_TYPE_GRAY && bit_depth < 8) {
         png_set_expand_gray_1_2_4_to_8(png_struct_ptr);
+    } else if (color_type == PNG_COLOR_TYPE_RGB || color_type == PNG_COLOR_TYPE_RGB_ALPHA) {
```

Comment:
This is a regression. The code no longer transforms RGB files into RGBA.

We need the set_filler call to add the alpha channel, and this is now in the else statement when color_type is PNG_COLOR_TYPE_RGB.